### PR TITLE
[MRG] Add warnings for firwin2 use in combination with ICA

### PR DESCRIPTION
**Pull request by cbrnr**
See discussion in #4045

### CODE REVIEW

**Comment by agramfort** [comment]
no objection to merge

File: mne/preprocessing/ica.py [hunk 1]
```diff
@@ -137,6 +137,20 @@ class ICA(ContainsMixin):
...
               enhancing reproducibility and stability of results.
 
+    .. warning:: ICA is sensitive to low-frequency drifts and therefore
+                 requires the data to be high-pass filtered prior to fitting.
+                 Typically, a cutoff frequency of 1 Hz is recommended. Note
+                 that FIR filters prior to MNE 0.15 used the 'firwin2' design
+                 method, which generally produces rather shallow filters that
+                 might not work for ICA processing. Therefore, it is
+                 recommended to use IIR filters for MNE up to 0.14. In MNE
+                 0.15, FIR filters can be designed with the `firwin` method,
```
Comment (Eric89GXL) [inline mne/preprocessing/ica.py]: double-backticks. And you use quotes above, might as well be consistent while you're changing it

File: mne/preprocessing/ica.py [hunk 2]
```diff
@@ -137,6 +137,20 @@ class ICA(ContainsMixin):
...
+                 which generally produces much steeper filters. This method
+                 will be the default FIR design method in MNE 0.16. In MNE
+                 0.15, you need to explicitly set 'fir_design=firwin' to use
```
Comment (Eric89GXL) [inline mne/preprocessing/ica.py]:
```
``fir_design='firwin'``
```

**Comment by cbrnr** [comment]
Done

**Comment by larsoner** [comment]
CircleCI failure is unrelated, merging -- thanks @cbrnr

**Pull request merged**